packages/core-mobile/app/screens/send/components/SendPVM.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useEffect, useMemo } from 'react'
+import { Account, useSendContext } from 'contexts/SendContext'
+import { Network } from '@avalabs/core-chains-sdk'
+import { TokenWithBalancePVM } from '@avalabs/vm-module-types'
+import { useSelector } from 'react-redux'
+import { selectActiveAccount } from 'store/account'
+import NetworkService from 'services/network/NetworkService'
+import { Contact, CorePrimaryAccount } from '@avalabs/types'
+import { getAddressXP } from 'store/utils/account&contactGetters'
+import { PVMSendOptions } from '../utils/types'
+import usePVMSend from '../hooks/usePVMSend'
+import SendTokenForm from './SendTokenForm'
+
+const SendPVM = ({
+  nativeToken,
+  network,
+  account,
+  setToAccount,
+  onOpenQRScanner,
+  onOpenAddressBook,
+  onSuccess,
+  onFailure
+}: {
+  nativeToken: TokenWithBalancePVM
+  network: Network
+  account: CorePrimaryAccount
+  setToAccount: (toAccount: Account) => void
+  onOpenQRScanner: () => void
+  onOpenAddressBook: () => void
+  onSuccess: (txHash: string) => void
+  onFailure: (txError: unknown) => void
+}): JSX.Element => {
+  const {
+    toAccount,
+    amount,
+    token,
+    setToken,
+    maxAmount,
+    error,
+    isValid,
+    isValidating,
+    isSending,
+    maxFee
+  } = useSendContext()
+  const provider = useMemo(
+    () => NetworkService.getAvalancheProviderXP(!!network.isTestnet),
+    [network]
+  )
+
+  const activeAccount = useSelector(selectActiveAccount)
+
+  const fromAddress = activeAccount?.addressPVM ?? ''
+
+  const { validate, send } = usePVMSend({
+    network,
+    fromAddress,
+    provider,
+    maxFee,
+    nativeToken,
+    account
+  })
+
+  useEffect(() => {
+    validate({
+      token,
+      toAddress: toAccount.address,
+      amount: amount?.bn ?? 0n
+    } as PVMSendOptions)
```

Comment:
I find it a bit strange that we need to manually pass these params to validate() when validate() inside usePVMSend already has access to all of the data from the send context. could we move this useEffect inside usePVMSend?

Reply:
good point. I'm still using `SendContext` so all the parameters for `validate` and `send` are accessible in `useXXXSend` hooks. moved `validate()` to `useEffect` in `useXXXSend` hooks. also removed the parameters of `validate` and `send` functions.

Reply:
amazing!!